### PR TITLE
[Fix] Fix backward typo in `mul_relu`

### PR DESCRIPTION
**Pull request by KEKE046**
https://github.com/SiriusNEO/Triton-Puzzles-Lite/blob/3868e24cffd0f791764fd0cc5c488debfd422b67/puzzles.py#L337-L347

`x.clone()` is a torch autograd function, we should use `x = x.clone().detach().requires_grad_(True)`

### CODE REVIEW

**Comment by SiriusNEO** [comment]
LGTM!
